backend/src/test/java/com/redhat/cloud/notifications/routers/sources/SecretUtilsTest.java

```diff
@@ -0,0 +1,511 @@
+    }
+
+    /**
+     * Tests that when the user provides a {@code null} "basic authentication" and "secret token" secret for an
+     * endpoint which has secrets stored in Sources, the deletion process for those secrets is triggered.
+     */
+    @Test
+    void updateSecretsForEndpointDeleteTest() {
+        // Create an endpoint that contains the expected data by the function under test.
+        Endpoint endpoint = new Endpoint();
+        WebhookProperties webhookProperties = new WebhookProperties();
+
+        // Simulate that we are sending an empty "basic authentication" and "secret token", but that we have those
+        // IDs on the database. This should trigger the deletion of these secrets on Sources.
+        webhookProperties.setBasicAuthenticationSourcesId(BASIC_AUTH_SOURCES_ID);
+        webhookProperties.setSecretTokenSourcesId(SECRET_TOKEN_SOURCES_ID);
+
+        endpoint.setProperties(webhookProperties);
+
+        // Call the function under test.
+        this.secretUtils.updateSecretsForEndpoint(endpoint);
+
+        // The IDs of the properties should be null now.
+        Assertions.assertNull(webhookProperties.getBasicAuthenticationSourcesId(), "the basic authentication's Sources ID isn't null");
+        Assertions.assertNull(webhookProperties.getSecretTokenSourcesId(), "the secret token's Sources ID isn't null");
+
+        // It should have triggered two "delete" calls to Sources to delete both of the secrets.
+        final int wantedNumberOfInvocations = 1;
+        Mockito.verify(this.sourcesServiceMock, Mockito.times(wantedNumberOfInvocations)).delete(Mockito.eq(BASIC_AUTH_SOURCES_ID));
+        Mockito.verify(this.sourcesServiceMock, Mockito.times(wantedNumberOfInvocations)).delete(Mockito.eq(SECRET_TOKEN_SOURCES_ID));
+    }
+
+    /**
+     * Tests that when the client updates an endpoint, but there are no secrets stored in Sources, or no valid secrets
+     * provided by the client, basically a NOP happens. For a valid secret we understand it as a secret token that is
+     * not blank.
+     */
+    @Test
+    void updateSecretsForEndpointNopTest() {
+        // Create an endpoint that contains the expected data by the function under test.
+        Endpoint endpoint = new Endpoint();
+        WebhookProperties webhookProperties = new WebhookProperties();
+        endpoint.setProperties(webhookProperties);
+
+        this.secretUtils.updateSecretsForEndpoint(endpoint);
+
+        final int wantedNumberOfInvocations = 0;
+        Mockito.verify(this.sourcesServiceMock, Mockito.times(wantedNumberOfInvocations)).getById(Mockito.anyLong());
+        Mockito.verify(this.sourcesServiceMock, Mockito.times(wantedNumberOfInvocations)).create(Mockito.any());
+        Mockito.verify(this.sourcesServiceMock, Mockito.times(wantedNumberOfInvocations)).update(Mockito.anyLong(), Mockito.any());
+        Mockito.verify(this.sourcesServiceMock, Mockito.times(wantedNumberOfInvocations)).delete(Mockito.anyLong());
```

Comment:
You could use this here and at a few other places in this class:
```suggestion
        Mockito.verifyNoInteractions(sourcesServiceMock);
```

Reply:
Much better. Refactored!